File: Queue/C/Queue.c
#include<stdio.h>
#include<stdlib.h>

struct node 
{
    int data;
    struct node * next;
};

typedef struct node NODE;
typedef struct node * PNODE;
typedef struct node ** PPNODE;

void enqueue(PPNODE first, int no)
{
    PNODE newn = NULL;
    PNODE temp = NULL;

    newn = (PNODE)malloc(sizeof(NODE));
    newn->data = no;
    newn->next = NULL;

    if (*first == NULL)
    {
        *first = newn;
    }
    else
    {
        temp = *first;

        while (temp->next != NULL)
        {
            temp = temp->next;
        }

        temp->next = newn;
    }
}

int dequeue(PPNODE first)
{
    PNODE temp = NULL;
    int value = 0;

    if(*first == NULL)
    {
        printf("Unable to delete Queue is empty\n");
        return -1;
    }
    else
    {
        temp = *first;
        value = temp->data;

        *first = (*first)->next;
        free(temp);

        return value;
    }   
}

void Display(PNODE first)
{
    if(first == NULL)
    {
        printf("Queue is empty\n");
        return;
    }

    while (first != NULL)
    {
        printf("| %d | - ",first->data);
        first = first->next;
    }
    
    printf("NULL\n");
}

int Count(PNODE first)
{
    int iCount = 0;

    if(first == NULL)
    {
        printf("Queue is empty\n");
        return 0;
    }

    while (first != NULL)
    {
        iCount++;
        first = first->next;
    }

    return iCount;
}

int main()
{
    PNODE head = NULL;
    int iRet = 0;
    int choice = 0;
    int value = 0;

    printf("\n=============================================================\n");
    printf("                       Simple Queue Manager                    \n");
    printf("=============================================================\n");

    while(1)
    {
        printf("\n-------------------------------------------------------------\n");
        printf("Please select an operation:\n");
        printf("-------------------------------------------------------------\n");
        printf(" 1 : Enqueue at rear\n");
        printf(" 2 : Dequeue from front\n");
        printf(" 3 : Display from front to rear\n");
        printf(" 4 : Count total elements\n");
        printf(" 0 : Exit\n");
        printf("-------------------------------------------------------------\n");

        printf("Enter your choice: \n");
        scanf("%d",&choice);

        if(choice == 1)
        {
            printf("Enter element to enqueue at rear: \n");
            scanf("%d",&value);
            enqueue(&head, value);
        }
        else if(choice == 2)
        {
            value = dequeue(&head);
            if(value != -1)
            {
                printf("Dequeued from front: %d\n",value);
            }
        }
        else if(choice == 3)
        {
            Display(head);
        }
        else if(choice == 4)
        {
            iRet = Count(head);
            printf("Total elements in deque: %d\n",iRet);
        }
        else if(choice == 0)
        {
            printf("Thank you for using our Application!\n");
            break;
        }
        else
        {
            printf("Invalid choice. Please try again.\n");
        }
    }

    return 0;
}

File: Queue/Python/Queue.py

class node:

    def __init__(self, no):
        self.data = no
        self.next = None

class Queue:

    def __init__(self):
        self.first = None
        self.iCount = 0

    def enqueue(self, no):

        newn = node(no)

        if(self.first == None):
            self.first = newn

        else:
            temp = self.first

            while(temp.next != None):
                temp = temp.next

            temp.next = newn

        self.iCount += 1

    def dequeue(self):

        if(self.first == None):
            print("Unable to delete Queue is empty")

        else:
            temp = self.first
            iValue = temp.data

            self.first = self.first.next
            
            del temp

            self.iCount -= 1
            return iValue

    def Display(self):

        if(self.first == None):
            print("Queue is empty")
            return

        temp = self.first

        while(temp != None):
            print(f"| {temp.data} | -> ",end="");
            temp = temp.next

        print("None")

    def Count(self):
        return self.iCount

def main():

    obj = Queue()

    print("\n=============================================================");
    print("                       Simple Queue Manager                    ");
    print("=============================================================");

    while True:
        print("\n-------------------------------------------------------------")
        print("Please select an operation:")
        print("-------------------------------------------------------------")
        print(" 1 : Enqueue at rear")
        print(" 2 : Dequeue from front")
        print(" 3 : Display from front to rear")
        print(" 4 : Count total elements")
        print(" 0 : Exit")
        print("-------------------------------------------------------------")

        choice = int(input("Enter your choice : "))

        if choice == 1:
            value = int(input("Enter element to enqueue at rear: "))
            obj.enqueue(value)

        elif choice == 2:
            iRet = obj.dequeue()
            if iRet is not None:
                print(f"Dequeued from front: {iRet}")

        elif choice == 3:
            obj.Display()

        elif choice == 4:
            print(f"Total elements in Queue: {obj.Count()}")

        elif choice == 0:
            print("Thank you for using our Application!")
            break

        else:
            print("Invalid choice, please try again")


if __name__ == "__main__":
    main()